src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java

```diff
@@ -1,35 +1,53 @@
 package uk.gov.hmcts.ccd.domain.types;
 
+import com.fasterxml.jackson.databind.JsonNode;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
-import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.beans.factory.annotation.Qualifier;
+import uk.gov.hmcts.ccd.ApplicationParams;
+import uk.gov.hmcts.ccd.data.definition.CachedCaseDefinitionRepository;
+import uk.gov.hmcts.ccd.data.definition.CaseDefinitionRepository;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.Category;
 
 import javax.inject.Named;
 import javax.inject.Singleton;
-
 import java.util.Collections;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-
-import uk.gov.hmcts.ccd.ApplicationParams;
-import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
-
+import java.util.stream.Collectors;
 
 @Named("DocumentValidator")
 @Singleton
 public class DocumentValidator implements BaseTypeValidator {
     public static final String TYPE_ID = "Document";
     static final String DOCUMENT_URL = "document_url";
     private static final String DOCUMENT_BINARY_URL = "document_binary_url";
+    private static final String CATEGORY_ID = "category_id";
+    private static final String UPLOAD_TIMESTAMP = "upload_timestamp";
+    private static final String NOT_TEXT_OR_EMPTY = " is not a text value or is empty";
 
     private static final Logger LOG = LoggerFactory.getLogger(DocumentValidator.class);
 
     private final ApplicationParams applicationParams;
-
-    public DocumentValidator(ApplicationParams applicationParams) {
+    private final TextValidator textValidator;
+    private final DateTimeValidator dateTimeValidator;
+    private final CaseDefinitionRepository caseDefinitionRepository;
+    private List<ValidationResult> validationResults;
+    private List<ValidationResult> isNullOrEmptyResult;
```

Comment:
These could cause problems being defined as class variables - can we move to local variables inside the `validate()` method please. Happy to have a chat to explain!